conans/test/integration/test_pkg_signing.py

```diff
@@ -0,0 +1,43 @@
+import os
+import textwrap
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+from conans.util.files import save
+
+
+def test_pkg_sign():
+    c = TestClient(default_server_user=True)
+    c.save({"conanfile.py": GenConanfile("pkg", "0.1").with_exports("export/*")
+            .with_exports_sources("export_sources/*").with_package_file("myfile", "mycontents!"),
+            "export/file1.txt": "file1!",
+            "export_sources/file2.txt": "file2!"})
+    signer = textwrap.dedent(r"""
+        import os
+
+        def sign(ref, files, folder):
+            print("Signing ref: ", ref)
+            print("Signing files: ", files)
+            signature = os.path.join(folder, "signature.asc")
+            open(signature, "w").write("\n".join(files))
+            return ["signature.asc"]
+
+        def verify(ref, files, folder):
```

Comment:
``files`` is the list of files, excluding already the metadata, and they are absolute paths, while ``folder`` is the ``download/metatada`` folder

Reply:
This interface looks indeed weird. I don't like the `folder` argument. I might prefer a `metadata_files` with the absolute paths, but I see that it would be confusing to introduce the `metadata` concept when the plugin is only about signing. But maybe this plugin could always write the returned files to `metadata/signing`, and then pass the `signed_files` instead of `folder`.

Reply:
It can be named ``metadata_folder`` instead of ``folder``? If passed a list of files, then users would need to iterate or find their wanted files in that list, instead of ``os.path.join(metadata_folder, "myfilename")``?

Other alternative is just passing the parent folder (this is the "download" one), which will contain:
- "donwnload" folder
   - conanfile.py
   - conanmanifest.txt
   - conan_sources.tgz
   - metadata
      - signature.asc
And then let the plugin itself should know the layout and relative ``metadata/signature.asc`` path?